lib/ansible/galaxy/dependency_resolution/providers.py

```diff
@@ -211,11 +212,45 @@ def find_matches(self, requirements):
         first_req = requirements[0]
         fqcn = first_req.fqcn
         # The fqcn is guaranteed to be the same
-        coll_versions = self._api_proxy.get_collection_versions(first_req)
+        version_req = "A SemVer-compliant version or '*' is required. See https://semver.org to learn how to compose it correctly. "
+        version_req += "This is an issue with the collection."
+        try:
+            coll_versions = self._api_proxy.get_collection_versions(first_req)
+        except TypeError as exc:
+            if first_req.is_concrete_artifact:
+                # Non hashable versions will cause a TypeError
+                raise ValueError(
+                    f"Invalid version found for the collection '{first_req}'. " + version_req
```

Comment:
Why don't you use f-string for both injected vars?
```suggestion
                    f"Invalid version found for the collection '{first_req}'. {version_req}"
```